Fix holding config variable in database through proxy objects

bogdanbodnar opened the pull request:
A config variable may not necessarily be None when not configured. Can be stored in the database with the help of proxy objects, which can  return the None value when accessed.
For example, to integrate with jazzband/django-constance.

bittner commented:
Unfortunately, I'd guess due to the change, tests are failing for the UserVoice template tag (see below).

This is related to the value of `USERVOICE_WIDGET_KEY=''` now probably being evaluated differently in [test_empty_key](https://github.com/jcassee/django-analytical/blob/master/analytical/tests/test_tag_uservoice.py#L43) and [test_overridden_empty_key](https://github.com/jcassee/django-analytical/blob/master/analytical/tests/test_tag_uservoice.py#L49).

Can you please fix this appropriately, so that the tests pass again? -- Thank you!

```python
======================================================================
ERROR: test_empty_key (analytical.tests.test_tag_uservoice.UserVoiceTagTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/jcassee/django-analytical/.tox/py27-django19/lib/python2.7/site-packages/django/test/utils.py", line 181, in inner
    return test_func(*args, **kwargs)
  File "/home/travis/build/jcassee/django-analytical/analytical/tests/test_tag_uservoice.py", line 43, in test_empty_key
    r = UserVoiceNode().render(Context())
  File "/home/travis/build/jcassee/django-analytical/analytical/templatetags/uservoice.py", line 53, in __init__
    WIDGET_KEY_RE, "must be an alphanumeric string")
  File "/home/travis/build/jcassee/django-analytical/analytical/utils.py", line 25, in get_required_setting
    raise AnalyticalException("%s setting is not set" % setting)
AnalyticalException: USERVOICE_WIDGET_KEY setting is not set
```
```python
======================================================================
ERROR: test_overridden_empty_key (analytical.tests.test_tag_uservoice.UserVoiceTagTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/jcassee/django-analytical/.tox/py27-django19/lib/python2.7/site-packages/django/test/utils.py", line 181, in inner
    return test_func(*args, **kwargs)
  File "/home/travis/build/jcassee/django-analytical/analytical/tests/test_tag_uservoice.py", line 49, in test_overridden_empty_key
    r = UserVoiceNode().render(Context(vars))
  File "/home/travis/build/jcassee/django-analytical/analytical/templatetags/uservoice.py", line 53, in __init__
    WIDGET_KEY_RE, "must be an alphanumeric string")
  File "/home/travis/build/jcassee/django-analytical/analytical/utils.py", line 25, in get_required_setting
    raise AnalyticalException("%s setting is not set" % setting)
AnalyticalException: USERVOICE_WIDGET_KEY setting is not set
```

bogdanbodnar commented:
OK, I fixed `test_empty_key`, rewritten it as similar tests in other nodes.

I deleted `test_overridden_empty_key` because all nodes are implemented so that they can not be created until there is no configuration. I think that key substitution in the context after creating the node doesn't make sense. Confirmation of this is lack of similar tests in other modules.

Maybe I just did not have enough time to understand the work of this library enough. If so, please explain in what the problem is  :)

jcassee commented:
Thanks, Bodnar!
